### PR TITLE
Updating additional methods for empty datatype filters

### PR DESCRIPTION
**Pull request by mineralntl**
These changes are tied to [datawave#1814](https://github.com/NationalSecurityAgency/datawave/pull/1814) as prerequisites for fully removing UniversalSet and handling empty but non-null datatype filters.

### CODE REVIEW

**Review by nonessentialprototype — APPROVED** [review]

File: src/main/java/datawave/query/util/MetadataHelper.java [hunk 1]
```diff
@@ -322,16 +322,12 @@ public Set<String> getAllFields(Set<String> ingestTypeFilter) throws TableNotFou
...
-        } else {
-            for (String datatype : ingestTypeFilter) {
-                fields.addAll(allFields.get(datatype));
-            }
-        }
+        fields.addAll(getFields(allFields, ingestTypeFilter));
         
         // Add any additional fields that are created at evaluation time and are hence not in the metadata table.
-        fields.addAll(evaluationOnlyFields);
+        if (!evaluationOnlyFields.isEmpty()) {
```
Comment (ivakegg) [inline src/main/java/datawave/query/util/MetadataHelper.java:328]: no need to check for empty here, but ok.

**Review by cogross — APPROVED** [review]

**Review by apmoriarty — APPROVED** [review]

File: src/main/java/datawave/query/util/MetadataHelper.java (hunk 1 above)
Comment (jwomeara) [reply src/main/java/datawave/query/util/MetadataHelper.java:328]: +1

**Review by jwomeara — APPROVED** [review]
aside from the unnecessary empty check, lgtm

**Review by jwomeara — CHANGES_REQUESTED** [review]
aside from the unnecessary empty check, lgtm

**Review by jwomeara — APPROVED** [review]

**Pull request merged**